### ISSUE TITLE
[Site Search] : Getting "Failure: null" message after typing invalid alias

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#17569]
Found when working on Support: [99630](https://dotcms.zendesk.com/agent/tickets/99630)
Reproducible in dotCMS 5.1.1, and Demo Site (5.2.0)

When you create a Site Search index, two input parameters are required: (1) Alias, and (2) Shards. If the alias you type in is not valid, an error is logged in the log file. However, the user will only see an error message stating: `Failure: null`

Part of the Java Stack Trace in the log is:
```
...
...
10:35:30.465  ERROR business.ESIndexAPI - Invalid alias name [test 1]: must not contain the following characters [ , ", *, \, <, |, ,, >, /, ?]
org.elasticsearch.indices.InvalidAliasNameException: Invalid alias name [test 1]: must not contain the following characters [ , ", *, \, <, |, ,, >, /, ?]
	at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService.validateIndexOrAliasName(MetaDataCreateIndexService.java:166) ~[elasticsearch-6.1.3.jar:6.1.3]
...
...
```
The error message must include the list of characters that are not allowed in the alias.

### Steps to reproduce the behavior:
1. Go to the **Dev Tools > Site Search** portlet.
2. Click the **Create Site Search Index** button.
3. In the **Alias** field, type in the following: "test 1", and click the **Create Site Search Index** button.
4. You'll see a modal window stating: `Failure: null`, and a Java Stack Trace just like the one above.

### Expected behavior
The error message in the modal windo must provide useful information to the user so that they can fix the problem.

## Acceptance Criteria

1. Match the design
2. Work in all the supported browsers (don't forget IE11 and iPad)
3. Multilanguage
4. Unit test

**Comment by jcastro-dotcms** [comment]
This issue is still present. Reported by another customer: [101402](https://dotcms.zendesk.com/agent/tickets/101402)

**Comment by bryanboza** [comment]
Now we are validating the special characters before send the data, but if you leave a space we are getting the same error mentioned above. 

Maybe we should to trim() the text and auto remove the blank spaces. 

We need some work here.

**Comment by bryanboza** [comment]
Fixed, tested on release-21.10 // docker // FF

**Comment by bryanboza** [comment]
We need add the same validation to the "Alias" filed in the "Job Scheduler" tab because right now we are overriding whatever you put after the space and if already exist we are overriding the existing one.

**Comment by alfredo-dotcms** [comment]
![image](https://user-images.githubusercontent.com/37185433/135360806-82d4e124-d83a-4713-acef-7838c118befe.png)

looks good!

**Comment by bryanboza** [comment]
After the last fix, now I have other problems: 

- We need add the blank spaces are not allowed as part of the message
- If you click in the Execute button, when the error message is present, we are broking something and this button does not working anymore. 
![image](https://user-images.githubusercontent.com/2641437/135540470-86f756e3-990a-4d53-91a1-199204dfc0e4.png)
- The same thing from the main create index modal (If you try to create when the message is displayed we broke something)
![image](https://user-images.githubusercontent.com/2641437/135540619-513d8db1-05c2-4e38-89f0-8b36b9c1a6a7.png)
- We need to register the error in the log file, after this fix now we are just showing the error at UI leven, but we are not printing the line of error at log level.

**Comment by nollymar** [comment]
**Internal QA:** Passed
<img width="1670" alt="Screen Shot 2021-10-04 at 12 20 30 PM" src="https://user-images.githubusercontent.com/8741395/135895819-937fd0bd-2f6d-425c-b52d-b0c44168f9bb.png">

**Comment by bryanboza** [comment]
Fixed, tested on release-21.10 // Docker // FF
